server/remote_cache/action_cache_server/action_cache_server.go

```diff
@@ -205,16 +205,25 @@ func (s *ActionCacheServer) GetActionResult(ctx context.Context, req *repb.GetAc
 		if err != nil {
 			return nil, err
 		}
+
+		// NOTE: To avoid double-counting AC hits, callers that specify a
+		// cached_action_result_digest don't do hit tracking on their own.  This
+		// means we need to track the full response size here instead.
+		uncompressedResultSizeBytes = int64(proto.Size(rsp))
+		isCacheHit = true
+		// Now that we've tracked size, wipe out the response.
```

Comment:
This is my fault for doing the tracking in a `defer` in the first place, but this code feels a little more error-prone / confusing now that we're setting the tracking variables in more than one place - might be worth doing the tracking explicitly, using a wrapper function like `func getLocallyCachedActionResult(...) { digest, isCacheHit, trackedSize, res, err := getLocallyCachedActionResultImpl(); if isCacheHit { track(...) }; return digest, res, err }`

Reply:
okay, the change is a little bit involved, but this looks about right to me now (and the tests still pass / it still works) for my sake, though, please take another look